rlberry/agents/common/models.py

```diff
@@ -0,0 +1,524 @@
+import torch
+import torch.nn as nn
+from torch.distributions import Categorical
+import numpy as np
+from gym import spaces
+from torch.nn import functional as F
+
+from rlberry.utils.configuration import Configurable
+
+
+class ValueNet(nn.Module):
+    def __init__(self, state_dim, hidden_size=64):
+        super(ValueNet, self).__init__()
+        # critic
+        self.critic = nn.Sequential(
+            nn.Linear(state_dim, hidden_size),
+            nn.Tanh(),
+            nn.Linear(hidden_size, hidden_size),
+            nn.Tanh(),
+            nn.Linear(hidden_size, 1)
+        )
+
+    def forward(self, state):
+        state_value = self.critic(state)
+        return torch.squeeze(state_value)
+
+
+class PolicyNet(nn.Module):
+    def __init__(self, state_dim, action_dim, hidden_size=64):
+        super(PolicyNet, self).__init__()
+        # actor
+        self.actor = nn.Sequential(
+            nn.Linear(state_dim, hidden_size),
+            nn.Tanh(),
+            nn.Linear(hidden_size, hidden_size),
+            nn.Tanh(),
+            nn.Linear(hidden_size, action_dim)
+        )
+        self.softmax = nn.Softmax(dim=-1)
+
+    def forward(self, state):
+        action_probs = self.softmax(self.actor(state))
+        dist = Categorical(action_probs)
+        return dist
+
+    def action_scores(self, state):
+        action_scores = self.actor(state)
+        return action_scores
+
+
+class ActorCritic(nn.Module):
+    def __init__(self, state_dim, action_dim):
+        super(ActorCritic, self).__init__()
+        # actor
+        self.actor = nn.Sequential(
+            nn.Linear(state_dim, 64),
+            nn.Tanh(),
+            nn.Linear(64, 64),
+            nn.Tanh(),
+            nn.Linear(64, action_dim),
+            nn.Softmax(dim=-1)
+        )
+        # critic
+        self.critic = nn.Sequential(
+            nn.Linear(state_dim, 64),
+            nn.Tanh(),
+            nn.Linear(64, 64),
+            nn.Tanh(),
+            nn.Linear(64, 1)
+        )
+
+    def forward(self):
+        raise NotImplementedError
+
+    def act(self, state):
+        action_probs = self.actor(state)
+        dist = Categorical(action_probs)
+        action = dist.sample()
+
+        return action, dist.log_prob(action)
+
+    def evaluate(self, state, action):
+        action_probs = self.actor(state)
+        dist = Categorical(action_probs)
+
+        action_logprobs = dist.log_prob(action)
+        dist_entropy = dist.entropy()
+
+        state_value = self.critic(state)
+
+        return action_logprobs, torch.squeeze(state_value), dist_entropy
+
+
+class Net(nn.Module):
```

Comment:
Isn't this a duplicate of the class `MultiLayerPerceptron` ?

Reply:
Not exactly: MultiLayerPerceptron is more configurable (number of layers, type of activation, etc.) while Net is forced to have 2-layers Relu.

However, Net is used by your policy gradient algorithms, and I prefer keeping it as it is for this PR, so that the changes are only related to DQN with minimal modifications to other parts of the library (i.e. PG algorithms).
I suggest that in a future commit, we improve code sharing by removing Net and replace it with MultiLayerPerceptron everywhere it is used. 

The same must be done for the `Memory` classes, which is reimplemented by each algorithm (A2C, PPO, AVEC PPO, CEM, and DQN) with very similar code, while it should be common code instead. But let's merge them independently from this DQN feature.